File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        List<Integer> dp = new ArrayList<>();
        dp.add(1);
        dp.add(2);

        for(int i=2;i<n; i++) {
            dp.add(dp.get(i-1) + dp.get(i-2));
        }
        return dp.get(n-1);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        memo = {}
        def backtrack(curr):
            if curr<0:  return 0
            if curr==0: return 1
            if curr in memo:    return memo[curr]
            memo[curr] = backtrack(curr-1)+backtrack(curr-2)
            return memo[curr]

        return backtrack(n)
